### PR TITLE
feat(unifiedreport): support json format for rows and also html break

### PR DESCRIPTION
**Pull request by shaheemazmalmmd**
## Description

support json format for obligationstext in unified report.

## How to test

* go to customize page 
* Add common obligations in json format
```json
{
  "1.a":"All relevant licenses ",
  "1.b":"Add all copyrights",
  "1.c":"Add all acknowledgements"
}
```
* also for additional obligations and Obligation And Risk Assessment

<a href="https://gitpod.io/#https://github.com/fossology/fossology/pull/2588"></a>

### CODE REVIEW

File: src/unifiedreport/agent/reportStatic.php [hunk 1]
```diff
@@ -284,15 +285,47 @@ function assessmentSummaryTable(Section $section, $otherStatement, $heading)
...
+        }
+      }
+    } else {
+      $table->addRow($rowWidth);
+      $cell = $table->addCell($firstColLen);
+      if (strpos($obligationText, '<br>') !== false) {
+        $cell = $table->addCell($secondColLen);
+        $obligationText = explode('<br>', $obligationText);
+        foreach ($obligationText as $value) {
+          $cell->addText(htmlspecialchars($value), $rowTextStyleRight, "pStyle");
```
Comment (GMishx) [inline src/unifiedreport/agent/reportStatic.php:323]: A suggestion to fix the line breaks for text breaks.

```suggestion
      if (strpos($obligationText, '\n') !== false) {
        $cell = $table->addCell($secondColLen);
        $obligationText = explode('\n', $obligationText);
        foreach ($obligationText as $value) {
          $value = str_replace("\r", "", $value);
          $cell->addText(htmlspecialchars($value), $rowTextStyleRight, "pStyle");
```

File: src/unifiedreport/agent/reportStatic.php [hunk 2]
```diff
@@ -284,15 +285,47 @@ function assessmentSummaryTable(Section $section, $otherStatement, $heading)
...
+    $jsonObligationText = json_decode($obligationText, true);
+    if (json_last_error() === JSON_ERROR_NONE) {
+      foreach ($jsonObligationText as $key => $value) {
+        $table->addRow($rowWidth);
+        $cell = $table->addCell($firstColLen)->addText(htmlspecialchars($key), $rowTextStyleLeft, "pStyle");
+        if (strpos($value, '<br>') !== false) {
+          $cell = $table->addCell($secondColLen);
+          $value = explode('<br>', $value);
+          foreach ($value as $value1) {
+            $cell->addText(htmlspecialchars($value1), $rowTextStyleRight, "pStyle");
```
Comment (GMishx) [inline src/unifiedreport/agent/reportStatic.php:310]: Suggestion to fix the line breaks.

```suggestion
        if (strpos($value, '\n') !== false) {
          $cell = $table->addCell($secondColLen);
          $value = explode('\n', $value);
          foreach ($value as $value1) {
            $value1 = str_replace("\r", "", $value1);
            $cell->addText(htmlspecialchars($value1), $rowTextStyleRight, "pStyle");
```

**Review by GMishx — CHANGES_REQUESTED** [review]
Simple line break fix suggestions.

File: src/unifiedreport/agent/reportStatic.php (hunk 1 above)
Comment (GMishx) [reply src/unifiedreport/agent/reportStatic.php:323]: Ping @shaheemazmalmmd to send notification.

Comment (shaheemazmalmmd) [reply src/unifiedreport/agent/reportStatic.php:323]: Noted

**Review by GMishx — APPROVED** [review]
Tested, working as expected.

**Pull request merged**